application/src/main/java/org/togetherjava/tjbot/commands/moderation/temp/RevocableModerationAction.java

```diff
@@ -62,11 +46,9 @@
      * Handle a failure that might occur during revocation, i.e. execution of the action returned by
      * {@link #revokeAction(Guild, User, String)}.
      * 
-     * @param failure the failure to handle
+     * @param errorResponse the error that occurred
      * @param targetId the id of the user who is targeted by the revocation
-     * @return a classification of the failure, decides whether the surrounding flow will continue
-     *         to handle the error further or not
+     * @return whether it should log an unknown failure message or not
      */
-    @NotNull
-    FailureIdentification handleRevokeFailure(@NotNull Throwable failure, long targetId);
+    boolean handleRevokeFailure(@NotNull ErrorResponse errorResponse, long targetId);
```

Comment:
I dont like that you changed `Throwable` to `ErrorResponse`. Design-wise, you are taking away capabilities from the system here. For example, if there is a temporary moderation action that can throw a FooException during revocation, it can not handle it now anymore. It can only handle Discord-API responses now, but not custom issues from our code.

I would suggest you revert back this part and use `Throwable failure` again and move the `if (failure instanceof ErrorResponse errorResponse) { ... }` back into the implementations (the abstract class and the temp-ban).

Reply:
Doesn't JDA's `queue()` only throw `ErrorResponse`s?

Reply:
No. Anything can happen when it executes your code. For example, what if you have a `.map(...)` which throws some exception? It will be exposed here as well.

The exception you get here is not only the actual Discord/JDA exceptions but everything that could go wrong during the whole execution stage, including your transformations.

Reply:
Maybe we should have this handle all throwables, so that the code will be valid in the future?

Reply:
And thats precisely what I proposed above. Just undo your changes on this part, they did not make the code better - but worse.